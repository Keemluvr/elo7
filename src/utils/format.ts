/**
 * Capitalizes the first letter of the input string and converts the rest to lowercase.
 *
 * @param {string} string - The input string to be capitalized.
 * @return {string} The capitalized string.
 */
export const capitalize = (string: string) =>
  string.charAt(0).toUpperCase() + string.slice(1).toLowerCase();

/**
 * Capitalizes each word in a string, including words separated by hyphens.
 *
 * @param {string} string - The input string to be capitalized.
 * @return {string} The capitalized string with each word capitalized and words separated by hyphens preserved.
 */
export const capitalizeWords = (string: string) => {
  return string
    .split(" ")
    .map((word) => {
      if (word.includes("-")) {
        const parts = word.split("-");
        return parts.map((part) => capitalize(part)).join("-");
      }
      return capitalize(word);
    })
    .join(" ");
};

/**
 * Formats a location string into a city, state, and country.
 *
 * @param {string} location - The location string to format.
 * @return {string} The formatted location string.
 */
export const formatLocation = (location: string) => {
  const [city, state, country] = location.split(", ");
  if (!city || !country) return "";
  return `${capitalizeWords(city)}, ${capitalizeWords(country)}`;
};

/**
 * Replaces specific substrings in a given string with their corresponding uppercase versions.
 *
 * @param {string} string - The input string to be processed.
 * @return {string} The modified string with the special cases replaced.
 */
export const specialCases = (string: string) => {
  return string
    .replace(/fp&a/gi, "FP&A")
    .replace(/Android/gi, "Android")
    .replace(/Ios/gi, "iOS")
    .replace(/devops/gi, "DevOps")
    .replace(/Sre/gi, "SRE");
};
