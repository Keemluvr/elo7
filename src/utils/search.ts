interface Data<T> {
  [key: string]: T[];
}

/**
 * Searches the data for entries that match all the terms in the query.
 *
 * @param {Data<T>} data - The data to search through.
 * @param {string | null} query - The search query.
 * @param {(keyof T)[]} properties - The properties to search within.
 * @return {Data<T> | undefined} The filtered data based on the search query.
 */
export const searchByCombinedTerms = <T>(
  data: Data<T>,
  query: string | null,
  properties: (keyof T)[]
): Data<T> | undefined => {
  if (!query) return;

  const terms = query.toLowerCase().split(" ");
  const filteredData: Data<T> = {};

  for (const key in data) {
    const filteredEntries = data[key].filter((entry: T) => {
      const entryString = properties
        .map((prop) => (entry[prop] as string).toLowerCase())
        .join(" ");
      return terms.every((term) => entryString.includes(term));
    });

    if (filteredEntries.length > 0) {
      filteredData[key] = filteredEntries;
    }
  }
  return Object.keys(filteredData).length > 0 ? filteredData : undefined;
};
