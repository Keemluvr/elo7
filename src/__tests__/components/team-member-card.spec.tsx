import TeamMemberCard from "@/components/team-member-card";
import { render, screen } from "@testing-library/react";
import { verifyAttributes } from "../utils/verifyAttributes";

let container = {} as DocumentFragment;

describe("TeamMemberCard component", () => {
  const name = "john-doe";
  const alt = "John Doe";

  beforeEach(() => {
    const { asFragment } = render(<TeamMemberCard name={name} alt={alt} />);
    container = asFragment();
  });

  it("should render the component", () => {
    const containerElement = screen.getByRole("img").parentElement;
    expect(containerElement).toBeInTheDocument();
  });

  it("should render the wrapper with the correct attributes", () => {
    const containerElement = screen.getByRole("img").parentElement;
    const containerAttributes = [["class", "team-member-card"]];
    verifyAttributes(containerElement as HTMLElement, containerAttributes);
  });

  describe("Image", () => {
    it("should render an image", () => {
      const img = screen.getByAltText(alt);
      expect(img).toBeInTheDocument();
    });

    it("should render an image with the correct attributes", () => {
      const imgElement = screen.getByAltText(alt);

      const expectedSrc =
        "/_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=3840&q=100";
      const imageAttributes = [
        ["src", expectedSrc],
        ["loading", "lazy"],
        ["alt", alt],
        ["aria-label", `imagem membro da equipe: ${name}`],
        ["class", "team-member-card-image"],
      ];
      verifyAttributes(imgElement, imageAttributes);
    });
  });

  test("matches snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
