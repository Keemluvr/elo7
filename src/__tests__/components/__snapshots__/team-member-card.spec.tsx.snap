// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMemberCard component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="team-member-card"
  >
    <img
      alt="John Doe"
      aria-label="imagem membro da equipe: john-doe"
      class="team-member-card-image"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=3840&q=100"
      srcset="/_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=640&q=100 640w, /_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=750&q=100 750w, /_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=828&q=100 828w, /_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=1080&q=100 1080w, /_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=1200&q=100 1200w, /_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=1920&q=100 1920w, /_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=2048&q=100 2048w, /_next/image?url=%2Fimages%2Fteam%2Fjohn-doe.webp&w=3840&q=100 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
</DocumentFragment>
`;
